cmd/oras/manifest/push.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manifest
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"oras.land/oras/cmd/oras/internal/file"
+	"oras.land/oras/cmd/oras/internal/option"
+)
+
+type pushOptions struct {
+	option.Common
+	option.Remote
+
+	targetRef string
+	fileRef   string
+	mediaType string
+}
+
+func pushCmd() *cobra.Command {
+	var opts pushOptions
+	cmd := &cobra.Command{
+		Use:   "push name[:tag|@digest] file",
+		Short: "[Preview] Push a manifest to remote registry",
+		Long: `[Preview] Push a manifest to remote registry
+** This command is in preview and under development. **
+
+Example - Push a manifest to repository 'locahost:5000/hello' and tag with 'latest':
+  oras manifest push localhost:5000/hello:latest manifest.json
+
+  Example - Push an ORAS artifact manifest to repository 'locahost:5000/hello' and tag with 'latest':
+  oras manifest push localhost:5000/hello:latest oras_manifest.json --media-type application/vnd.cncf.oras.artifact.manifest.v1+json
```

Comment:
nit: it is better to put the flags before the arguments in the example.
```suggestion
  oras manifest push --media-type application/vnd.cncf.oras.artifact.manifest.v1+json localhost:5000/hello:latest oras_manifest.json
```